src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/ILGen.cs

```diff
@@ -574,10 +574,7 @@ internal static void EmitConstant(this ILGenerator il, object value, Type type)
             throw ContractUtils.Unreachable;
         }
 
-        internal static bool ShouldLdtoken(Type t)
-        {
-            return t is TypeBuilder || t.IsGenericParameter || t.GetTypeInfo().IsVisible;
-        }
+        private static bool ShouldLdtoken(Type t) => t.IsGenericParameter || t.GetTypeInfo().IsVisible;
```

Comment:
Yes, that's it. The following works in LINQPad and returns the created type, rather than the TypeBuilder used to create it:

```C#
AssemblyBuilder assembly = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("Name"), AssemblyBuilderAccess.Run);
ModuleBuilder module = assembly.DefineDynamicModule("Name");
var typeBuilder = module.DefineType("Type");
var meth = typeBuilder.DefineMethod("r", MethodAttributes.Public | MethodAttributes.Static, typeof(TypeBuilder), Type.EmptyTypes);
var e = Expression.Lambda<Func<Type>>(Expression.Constant(typeBuilder, typeof(Type)));
e.CompileToMethod(meth);
var t = typeBuilder.CreateTypeInfo().AsType();
object result = t.GetMethod("r").Invoke(null, new object[0]);
LINQPad.Extensions.Dump(result);
```

Need to allow that to continue.